packages/gator-permissions-snap/src/clients/nonceCaveatClient.ts

```diff
@@ -58,62 +65,98 @@ export class NonceCaveatClient {
       throw new InvalidInputError(message);
     }
 
-    // Check if we're on the correct chain and switch if not
-    const selectedChain = await this.#ethereumProvider.request<Hex>({
-      method: 'eth_chainId',
-      params: [],
-    });
-
-    if (selectedChain && hexToNumber(selectedChain) !== chainId) {
-      await this.#ethereumProvider.request<Hex>({
-        method: 'wallet_switchEthereumChain',
-        params: [{ chainId: numberToHex(chainId) }],
-      });
-
-      const updatedChain = await this.#ethereumProvider.request<Hex>({
-        method: 'eth_chainId',
-        params: [],
-      });
-
-      if (updatedChain && hexToNumber(updatedChain) !== chainId) {
-        throw new ChainDisconnectedError(
-          'Selected chain does not match the requested chain',
+    const { retries = 1, delayMs = 1000 } = retryOptions ?? {};
+
+    // Try up to initial attempt + retry attempts
+    for (let attempt = 0; attempt <= retries; attempt++) {
```

Comment:
should this be leveraging the `makeValidatedRequestWithRetry` function to avoid having to implement it's own retry mechanism?

Reply:
`makeValidatedRequestWithRetry` only works with `fetch`, while `nonceCaveatClient` and `blockchainMetadataClient` rely on `ethereumProvider` and therefore implement their own retry logic. We could consider refactoring this into a shared retry mechanism in a future PR.